files/views.py

```
@@ -2096,3 +2097,31 @@ def get(self, request, format=None):
+            raise Http404
+
+        if not media.is_encrypted or not media.encryption_key:
+            raise Http404
+
+        if not check_media_access_permission(request, media):
+            return HttpResponse(status=403)
+
+        try:
+            key_bytes = bytes.fromhex(media.encryption_key)
+        except ValueError:
+            raise Http404
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)